bitrisescript/src/bitrisescript/task.py

```diff
@@ -86,16 +102,27 @@ def get_bitrise_workflows(config: dict[str, Any], task: dict[str, Any]) -> list[
     return workflows
 
 
-def get_build_params(task: dict[str, Any]) -> dict[str, Any]:
+def get_build_params(task: dict[str, Any], workflow: str = None) -> list[dict[str, Any]]:
     """Get the build_params from the task payload or an empty dict.
 
     Args:
         task (dict): The task definition.
+        workflow (str): Optional workflow reference used to load workflow_params
 
     Returns:
-        dict: The bitrise build_params to specify. Empty dict if unspecified.
+        dict: The bitrise build_params to specify. Always adds workflow_id to the returned dict.
     """
-    return task["payload"].get("build_params", {})
+    global_params = task["payload"].get("global_params", {})
+    workflow_params = task["payload"].get("workflow_params", {}).get(workflow)
```

Comment:
Idea: What if instead of an object keyed by workflow id, this was just a flat list that included `workflow_id` in the params?

E.g, instead of:
```
workflow_params:
    foo: [{environments: {VAL: 1}, {environments: {VAL: 2}}]
    bar: []
```

We had:
```
workflow_params:
    - {workflow_id: foo, environment: {VAL: 1}}
    - {workflow_id: foo, environment: {VAL: 2}}
    - {workflow_id: bar}
```
?

I feel like this would make the logic a little bit cleaner, though if we do this we'd need to remember to validate the specified workflows against our scopes and error out if we're missing any scopes.